### PR TITLE
EPMLSTRCMW-223 Make Scalastyle more aggressive

### PR DESCRIPTION
**Pull request by Nikolai-Isaev**
chore: Make build fail on scalastyle warnings

- build now fails when scalastyle rules are violated
- scalastyle exceptions for "valid" existing warnings added

### CODE REVIEW

File: repositories/src/main/scala/cromwell/pipeline/datastorage/dao/mongo/DocumentRepository.scala [hunk 1]
```diff
@@ -11,12 +11,11 @@ import org.mongodb.scala.{ Document, MongoCollection }
 import scala.concurrent.{ ExecutionContext, Future }
 
 class DocumentRepository(collection: MongoCollection[Document])(implicit ec: ExecutionContext) {
-
-  //noinspection ScalaStyle
+  // scalastyle:off if.brace
```
Comment (myazinn) [inline repositories/src/main/scala/cromwell/pipeline/datastorage/dao/mongo/DocumentRepository.scala]: I suggest we remove this stupid rule

File: repositories/src/main/scala/cromwell/pipeline/datastorage/dto/Project.scala [hunk 2]
```diff
@@ -8,7 +8,7 @@ import play.api.libs.json.JsonNaming.SnakeCase
 import play.api.libs.json._
 import java.nio.file.{ Path, Paths }
 import slick.lifted.MappedTo
-
+// scalastyle:off number.of.types
```
Comment (myazinn) [inline repositories/src/main/scala/cromwell/pipeline/datastorage/dto/Project.scala:11]: please create a ticket to split this file

Comment (Nikolai-Isaev) [reply repositories/src/main/scala/cromwell/pipeline/datastorage/dto/Project.scala:11]: [Ticket](https://jirapct.epam.com/jira/browse/EPMLSTRCMW-303) created

File: repositories/src/main/scala/cromwell/pipeline/datastorage/dao/mongo/DocumentRepository.scala (hunk 1 above)
Comment (Nikolai-Isaev) [reply repositories/src/main/scala/cromwell/pipeline/datastorage/dao/mongo/DocumentRepository.scala]: Removed

**Review by myazinn — APPROVED** [review]

File: services/src/main/scala/cromwell/pipeline/service/AuthService.scala [hunk 3]
```diff
@@ -91,10 +91,11 @@ object AuthService {
         val refreshTokenContent = RefreshTokenContent(user.userId, None)
         getAuthResponse(accessTokenContent, refreshTokenContent, Instant.now.getEpochSecond)
       }
-
+      // scalastyle:off if.brace
```
Comment (myazinn) [inline services/src/main/scala/cromwell/pipeline/service/AuthService.scala]: Sorry, I've just noticed it. You need to remove it as well

Comment (Nikolai-Isaev) [reply services/src/main/scala/cromwell/pipeline/service/AuthService.scala]: Missed that, sry. Fixed

**Review by Dmitrii-Shevchenko — APPROVED** [review]

**Pull request merged**